### PR TITLE
Use only aptly and massive cleanup

### PR DESCRIPTION
**Pull request by mariogrip**
It's time! lets move to aptly only.

Do not merge this, just approve then I'll merge and refresh the aptly to match 100% whats on our current freight.

### CODE REVIEW

File: build-repo.sh [hunk 1]
```diff
@@ -17,78 +17,56 @@
...
-    export WORKSPACE="$rootwp/mbuild/$d"
+  for d in $MULTI_DIST ; do
+    echo "Repo-ing for $d"
+    release="$d"
+
+    WORKSPACE="$rootwp/mbuild/$d"
     cd "$WORKSPACE"
-    mkdir $BASE_PATH || true
+    
+    # Copy back for jenkins to archive
```
Comment (peat-psuwit) [inline build-repo.sh]: Doesn't archiving happen before this script is run?

https://github.com/ubports/build-tools/blob/30e43786855be735e37fa73428c796287e70fc78/Jenkinsfile#L58-L59

Comment (mariogrip) [reply build-repo.sh]: Huh it does indeed, i can remove that then :)

**Review by UniversalSuperBox — APPROVED** [review]
Without having my own replica of the build infrastructure, I'm unfortunately unable to give this a lot of testing. But it seems fine on static analysis.

File: build-repo.sh [hunk 2]
```diff
@@ -17,78 +17,41 @@
...
+    tar --overwrite -xvzf $t
+    rm $t || true
   done
 
-  if ! aptly -db-open-attempts=40 repo show $release ; then
-    aptly -db-open-attempts=40 repo create -distribution="$release" $release
-    aptly -db-open-attempts=40 publish repo $release filesystem:repo:main
-  fi
+  for d in $MULTI_DIST ; do
+    release="$d"
```
Comment (peat-psuwit) [inline build-repo.sh]: Can't we:

```suggestion
  for release in $MULTI_DIST ; do
```

?

File: build-repo.sh [hunk 3]
```diff
@@ -17,78 +17,41 @@
...
-	echo "Doing multibuild"
-	MULTI_DIST=$(cat multidist.buildinfo)
-  for t in multidist*.tar.gz ; do
-    tar --overwrite -xvzf $t
-  done
-	rm multidist*.tar.gz || true
-  export rootwp=$(pwd)
+push_aptly() {
+    local release="$1"
+    local path="$2"
```
Comment (peat-psuwit) [inline build-repo.sh:24]: `local` technically isn't in POSIX sh. Consider changing shebang to `/bin/dash`?

Comment (mariogrip) [reply build-repo.sh:24]: Fixed, changed to bash

File: build-repo.sh (hunk 2 above)
Comment (mariogrip) [reply build-repo.sh]: Fixed thanks

File: build-repo.sh (hunk 3 above)
Comment (peat-psuwit) [reply build-repo.sh:24]: I wrote `dash` not `bash` (`dash` does support `local`). But `bash` works too I guess (and maybe more portable? but you still need cowbuilder...). The startup speed gain is completely drowned out by other program's execution speed.

**Review by peat-psuwit — APPROVED** [review]
Seems OK now, except for a few quoting warnings in shellcheck which doesn't seem to be a problem.

**Comment by UniversalSuperBox** [comment]
This has been superseded by many PRs now, right @peat-psuwit?

**Comment by peat-psuwit** [comment]
I guess so. :)

**Pull request closed without merging**